components/calendar/calendar.ts

```diff
@@ -231,14 +236,8 @@ export class Calendar implements AfterViewInit,OnInit,OnDestroy,ControlValueAcce
     constructor(public el: ElementRef, public domHandler: DomHandler,public renderer: Renderer) {}
 
     ngOnInit() {
-        let today = new Date();
```

Comment:
Because they are not used anymore, both `weekDays` and `currentMonthText` are pulled directly from `locale`. `locale` has setter that recreates component with new locale.
I hope I haven't missed anything... in my case and in modified calendar demo it works well.